### PR TITLE
exchangeToken functions, code restructure

### PR DESCRIPTION
**Pull request by aaronli03**
## What
- exchangeToken function implemented
- handleExchangeTokenResponse implemented
- tests for the above two functions
- setup for Jest
- localStorage mock setup
- code restructure

## Why
- exchangeToken exchanges the code from authorization to retrieve access and refresh tokens for future requests
- handleExchangeTokenResponse is a callback function used to store tokens in local storage
- tests to ensure that functions are working as intended
- Jest is needed to mock local storage manipulation
- restructure for better organization

## Impact
- Will customer experience be significantly altered? No
- Does the code change cover more than one task? No

## Testing
- tested locally by running `npm run dev`

### CODE REVIEW

File: src/api/spotify/auth.ts [hunk 1]
```diff
@@ -46,3 +49,40 @@ export function login() {
...
+		grant_type: 'authorization_code',
+		client_id: clientId as string,
+		code_verifier: verifier as string
+	});
+	const xhr = new XMLHttpRequest();
+	xhr.open('POST', 'https://accounts.spotify.com/api/token', true);
+	xhr.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded');
+	xhr.setRequestHeader('Authorization', 'Basic ' + btoa(clientId + ':' + clientSecret));
+	xhr.send(body);
+	xhr.onload = handleExchangeTokenResponse;
```
Comment (byjackli) [inline src/api/spotify/auth.ts:67]: I thought we were using fetch, not xml?

File: src/api/spotify/auth.ts [hunk 2]
```diff
@@ -46,3 +49,40 @@ export function login() {
...
+}
+
+export function handleExchangeTokenResponse(this: XMLHttpRequest & MyResponseType): void {
```
Comment (byjackli) [inline src/api/spotify/auth.ts:70]: same comment as above

File: tests/api/auth.test.ts [hunk 3]
```diff
@@ -113,4 +120,22 @@ describe('Auth', () => {
 			expect(responseData.status).toBe(200);
 		};
 	});
+	it('exchangeToken localStorage', () => {
```
Comment (byjackli) [inline tests/api/auth.test.ts:123]: Can this message be a little more specific?

File: src/api/spotify/auth.ts (hunk 1 above)
Comment (aaronli03) [reply src/api/spotify/auth.ts:67]: Didn’t want too many changes on one or so was going to make one dedicated to change from using xhr to fetch

**Review by byjackli — APPROVED** [review]

**Pull request merged**